nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-processor-details.js

```diff
@@ -215,9 +215,10 @@
                     }
 
                     var executionNode = details.config['executionNode'];
+                    var executionNodeRestricted = details.executionNodeRestricted
 
                     // only show the execution-node when applicable
-                    if (nfClusterSummary.isClustered() || executionNode === 'PRIMARY') {
+                    if (nfClusterSummary.isClustered() && executionNodeRestricted !== true) {
```

Comment:
I think we need the executionNode === 'PRIMARY' to still be considered here as well.